misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java

```diff
@@ -110,30 +111,29 @@ public CoverageReport(Module module, Map<String, ModuleCoverage> moduleCoverageM
     public void generateReport(JBallerinaBackend jBallerinaBackend, String includesInCoverage) throws IOException {
         String orgName = this.module.packageInstance().packageOrg().toString();
         String packageName = this.module.packageInstance().packageName().toString();
-        String version = this.module.packageInstance().packageVersion().toString();
 
         List<Path> filteredPathList;
 
         if (!module.testDocumentIds().isEmpty()) {
             filteredPathList =
                     filterPaths(jBallerinaBackend.jarResolver().getJarFilePathsRequiredForTestExecution(
-                            this.module.moduleName()), jBallerinaBackend);
+                            this.module.moduleName()), jBallerinaBackend, module.packageInstance());
         } else {
             filteredPathList = filterPaths(jBallerinaBackend.jarResolver().getJarFilePathsRequiredForExecution(),
-                    jBallerinaBackend);
+                    jBallerinaBackend, module.packageInstance());
         }
-
         if (!filteredPathList.isEmpty()) {
-            addCompiledSources(filteredPathList, orgName, packageName, version, false,
-                    includesInCoverage);
-            execFileLoader.load(executionDataFile.toFile());
-            final CoverageBuilder coverageBuilder = analyzeStructure();
-            // Create Testerina coverage report
-            createReport(coverageBuilder.getBundle(title), moduleCoverageMap);
-            // Add additional dependency jars for Coverage XML if included
+            CoverageBuilder coverageBuilder = generateTesterinaCoverageReport(orgName, packageName, filteredPathList);
+            // Add additional dependency jars for Jacoco Coverage XML if included
             if (includesInCoverage != null) {
-                addCompiledSources(getDependencyJarList(jBallerinaBackend), orgName, packageName, version,
-                       true, includesInCoverage);
+                List<Path> dependencyPathList = getDependencyJarList(jBallerinaBackend);
```

Comment:
Shall we move getting the jar list to a separate method?

Reply:
Fixed with 1fabfc7e25ef44de12795d3e47d8a39b0d2efede